### PR TITLE
Contributing guidelines.

### PR DESCRIPTION
**Pull request by anmacdonald**
Here is my first draft of the guidelines. 

@mrwillyees Not asking you for a review, but as someone who has recently onboarded with the toolkit, any information you think is worth adding would be greatly appreciated.  😄

### CODE REVIEW

File: calcite-qml/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+1. The developer, in their application, chooses to import a control, e.g. `Button`.
+2. QML checks the currently applied style and imports the specific `Button.qml` from within that style folder.
+3. The specific `Button.qml` provides the look-and-feel of the button. This inherits from `T.Button.qml`, a [Qt Quick template](https://doc.qt.io/qt-5/qtquicktemplates2-index.html) which handles the business logic for the `Button` type.
+
+In this module, We are primarily interested in step 3. We provide our own custom `Button.qml`, `Slider.qml`, etc. which lay the UX on top of `T.Button.qml` and `T.Slider.qml` respectively.
+
+All UI elements which inherit from [Control](https://doc.qt.io/qt-5/qml-qtquick-controls2-control.html) may
+be styled. As more controls are added to [this list](https://doc.qt.io/qt-5/qtquick-controls2-qmlmodule.html), we 
+should be proactive in providing a Calcite equivalent in a timely manner. Your contributions are welcome here!
+
```
Comment (mrwillyees) [inline calcite-qml/CONTRIBUTING.md:46]: I would add info about the custom style in a project can be set by adding the file [qtquickcontrols2.conf](https://github.com/Esri/arcgis-runtime-toolkit-qt/blob/main/uitools/examples/cpp_quick/qtquickcontrols2.conf)
and that to make it work, it is needed to QQuickStyle::addStylePath("qrc:///esri.com/imports/"); into the main.
Got me off guard the first time I tried to set it up!

Comment (anmacdonald) [reply calcite-qml/CONTRIBUTING.md:46]: This should be covered under the Calcite README!

File: calcite-qml/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+- The [Calcite Web](https://esri.github.io/calcite-web/documentation/components) style guide.
+
+- Calcite [colors and themes](https://developers.arcgis.com/calcite-design-system/foundations/colors/).
+
+- [Calcite design system](https://developers.arcgis.com/calcite-design-system/).
+
+## Design philosophy 
+
+1. We are writing a custom QML style. Wherever there is a direct conflict between the QML 
+and Calcite styling systems, QML should be favoured.
```
Comment (ldanzinger) [inline calcite-qml/CONTRIBUTING.md:23]: 🙃  UK Spelling...

```suggestion
and Calcite styling systems, QML should be favored.
```

File: calcite-qml/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+2. Coherence is key. Favour designs which look cohesive
```
Comment (ldanzinger) [inline calcite-qml/CONTRIBUTING.md:25]:
```suggestion
2. Coherence is key. Favor designs which look cohesive 
```

**Review by mfeigl — APPROVED** [review]
LGTM - thanks for adding me to review!

**Pull request merged**